### PR TITLE
WIP Working on adding localkube to minikube-iso.  Need to add ability to …

### PR DESCRIPTION
**Pull request by aaron-prindle**
…version localkube, potentially add log rotation and service restart.

Ref: #830

### CODE REVIEW

**Comment by aaron-prindle** [comment]
After a discussion offline I am going to remove localkube from the iso but still add it as a systemd service

**Comment by aaron-prindle** [comment]
With systemd we can also add the following:
Automatic restarting of localkube on crash
Log rotation for systemd journal

Please leave comments with you opinon on adding these or if you have additional suggestion for systemd and the [Unit] file for localkube

**Comment by dlorenc** [comment]
+1, let's use it for restarting and log rotation.

**Comment by aaron-prindle** [comment]
PTAL, have moved all iso-checking logic into bash commands.

File: pkg/minikube/cluster/commands.go [hunk 1]
```diff
@@ -82,10 +116,82 @@ func GetStartCommand(kubernetesConfig KubernetesConfig) (string, error) {
...
+ExecReload=/bin/kill -s HUP $MAINPID
+
+[Install]
+WantedBy=multi-user.target
+`
+
+func GetStartCommandSystemd(kubernetesConfig KubernetesConfig) (string, error) {
+
+	flagVals := make([]string, len(constants.LogFlags))
+	for _, logFlag := range constants.LogFlags {
```
Comment (r2d4) [inline pkg/minikube/cluster/commands.go]: The flag parsing should be shared across both start commands as much as possible.  Maybe a command to generate `/usr/local/bin/localkube {{.Flags}} \
 +--generate-certs=false --logtostderr=true --enable-dns=false --node-ip={{.NodeIP}}`
and then use that in both templates.

File: pkg/minikube/cluster/commands.go [hunk 2]
```diff
@@ -28,17 +28,51 @@ import (
 )
 
 // Kill any running instances.
-var stopCommand = "sudo killall localkube || true"
 
-var startCommandTemplate = `
+var startCommandB2DTemplate = `
```
Comment (r2d4) [inline pkg/minikube/cluster/commands.go]: maybe this becomes with the localkube command templated out
```
sudo sh -c 'PATH=/usr/local/sbin:$PATH nohup {{.RunLocalkube}} > {{.Stdout}} 2> {{.Stderr}} < /dev/null & echo $! > {{.Pidfile}} &'
```

**Comment by aaron-prindle** [comment]
@minikube-bot retest this please

**Comment by aaron-prindle** [comment]
PTAL, have refactored to have single localkube template that both start commands use

**Pull request merged**